src/librbd/internal.cc

```diff
@@ -72,6 +73,19 @@ namespace librbd {
 
 namespace {
 
+struct C_CreateV2Image : public Context {
```

Comment:
The downside is that you need to handle the deletion of the dummy ImageCtx here and eventually within rbd-mirror (since it will invoke ```CreateRequest::create``` directly).